fg21sim/extragalactic/pointsources/pointsources.py

```diff
@@ -0,0 +1,361 @@
+# Copyright (c) 2016 Zhixian MA <zxma_sjtu@qq.com>
+# MIT license
+
+import os
+
+import numpy as np
+import pandas as pd
+import astropy.units as au
+import healpy as hp
+
+from .flux import Flux
+from .starforming import StarForming
+from .starbursting import StarBursting
+from .radioquiet import RadioQuiet
+from .fr1 import FRI
+from .fr2 import FRII
+
+
+class PointSources:
+    """
+    This class namely pointsource is designed to generate PS catelogues,
+    read csv format PS lists, calculate the flux and surface brightness
+    of the sources at different frequencies, and then ouput hpmaps
+
+    functions
+    ---------
+    read_csv
+        read the csv format files, judge the PS type and
+        transformed to be iterable numpy.ndarray.
+
+    calc_flux
+        calculate the flux and surface brightness of the PS.
+
+    draw_elp
+        processing on the elliptical and circular core or lobes.
+
+    draw_circle
+        processing on the circular star forming or bursting galaxies
+
+    draw_ps
+        generate hpmap with respect the imput PS catelogue
+"""
+
+    def __init__(self, configs):
+        self.configs = configs
+        self._get_configs()
+        self.files = []
+
+    def _get_configs(self):
+        """Load configs and set the attributes"""
+        # nside of the healpix cell
+        self.nside = self.configs.getn("common/nside")
+        # frequencies
+        self.freq = self.configs.getn("frequency/frequencies")
+        # save flag
+        self.save = self.configs.getn("extragalactic/pointsources/save")
+
+    def gen_catelogue(self):
+        """Generate the catelogues"""
+        # Init
+        sf = StarForming(self.configs)
+        sb = StarBursting(self.configs)
+        rq = RadioQuiet(self.configs)
+        fr1 = FRI(self.configs)
+        fr2 = FRII(self.configs)
+
+        # Save
+        if self.save:
+            self.files.append(sf.save_as_csv()[1])
+            self.files.append(sb.save_as_csv()[1])
+            self.files.append(rq.save_as_csv()[1])
+            self.files.append(fr1.save_as_csv()[1])
+            self.files.append(fr2.save_as_csv()[1])
+
+    def read_csv(self,filepath):
+        """
+        Read csv format point source files,judge its class
+        type according to its name.
+
+        Parameters
+        ----------
+        filepath: str
+            Path of the file.
+        """
+        # Split to folder name and file name
+        filename = os.path.basename(filepath)
+        # Split and judge point source type
+        class_list = ['SF', 'SB', 'RQ', 'FRI', 'FRII']
+        class_name = filename.split('.')[0]
+        ps_type = class_list.index(class_name) + 1
```

Comment:
Since the method `read_csv` is a general one, which loads `PS list` from outside csv files, requirement for `ps_type` is not intelligent. Besides, I don't think present strategy is time consuming.